volatility3/cli/__init__.py

```diff
@@ -463,10 +476,7 @@ def run(self):
         try:
             # Construct and run the plugin
             if constructed:
-                grid = constructed.run()
-                renderer = renderers[args.renderer]()
-                renderer.filter = text_filter.CLIFilter(grid, args.filters)
-                renderer.render(grid)
+                renderers[args.renderer](output_filename = args.output_filename).render(constructed.run())
```

Comment:
Can I ask why this change was included?  It doesn't seem relevant to the purpose of the PR, and it makes what's going on much harder for a person to parse.  What benefit does the change provide?

Reply:
This also doesn't adhere to the renderer interface.  `output_filename` isn't one of the parameters expected by the `Renderer` class and I'm surprised you can just access `output_filename` and have it work?  This should probably be a `RendererOption` (which were never fully fleshed out, but we probably should get round to doing at some point), which renderers can then choose to act on or not.

See https://github.com/volatilityfoundation/volatility3/blob/develop/volatility3/framework/interfaces/renderers.py#L38.

Reply:
You are right this change was not necessary. It was a dirty way, I juste add `args.output_filename` as constructor attribute. See https://github.com/volatilityfoundation/volatility3/pull/1137/commits/fa7e97aaa01f637862deb054237a50e5920ed731